lib/logwriter.c

```diff
@@ -1189,11 +1188,10 @@ log_writer_update_message_stats(LogWriter *self, const LogMessage *msg, gsize ms
       unix_time_set_now(&now);
       gint64 diff = unix_time_diff_in_msec(&now, &msg->timestamps[LM_TS_RECVD]);
 
-      if (self->last_delay_update != now.ut_sec)
+      if (stats_counter_get(self->metrics.message_delay_sample_age) != (gsize) now.ut_sec)
```

Comment:
this adds an atomic read in place of a non-atomic one, that's why I originally used a separate variable for this purpose.

this is always executing on a single thread, so we don't need the atomic variable and this is on the fastpath.

Reply:
I did not consider that an atomic read is undesired, I guess it has performance implications?

Anyways, fixed, thanks!

Reply:
It does have some, although it's less than an atomic write.